### ISSUE TITLE
custom-select class disapear after wrapping in PrependedText

### ISSUE THREAD

**Issue by iGoosiatin** [django-crispy-forms/django-crispy-forms#1111]
* Package version: 1.11.0
* Django version: 3.0.5
* Python version: 3.9.0
* Template pack: bootstrap4

### Description:

Generating custom layout, both 'owner' and 'currency' are selects
```
class AccountForm(forms.ModelForm):
    def __init__(self, *args, **kwargs):
        super(AccountForm, self).__init__(*args, **kwargs)
        self.helper = FormHelper(self)
        self.helper.form_show_labels = False
        self.helper.field_class = 'input-group-lg'

        # owner = PrependedText('owner', 'Owner:')
        currency = PrependedText('currency', 'Currency:')

        submit = Submit('submit', 'Submit')
        submit.field_classes = "btn btn-outline-dark btn-block btn-lg"

        self.helper.layout = Layout(

            'owner',
            currency,
            submit
        )

    class Meta:
        model = Account
        fields = ['owner', 'currency']
```
Generated HTML - no "custom-select" class for 'currency'
```
<select name="owner" class="select form-control custom-select" id="id_owner"> 
        <option value="3" selected="">Name3</option>
       <option value="2">Name2</option>
</select>
<select name="currency" class="select form-control" id="id_currency">
   <option value="€" selected="">EUR</option>
   <option value="$">USD</option>
</select>
```

**Comment by smithdc1** [comment]
Thanks 👍this is related to #1093. 

Let's keep this as I think this is a smaller part of the issue and we may be able to solve this bit a little easier. 🤷‍♀️

**Comment by cpina** [comment]
I should have time over the weekend to look at this :-) if anyone goes ahead before mention it here...

**Comment by cpina** [comment]
I'm thinking of a better solution (I'm thinking of a way to refactor the 4 current `{% crispy_field ... %}` calls in `field.html` but this works: https://github.com/cpina/django-crispy-forms/commit/6b2df6c1e1b295d8bccb5213e895de30f1c48c1b

Refactoring two of them could be done adding a new template and including it. I'll think a bit more.

**Comment by smithdc1** [comment]
@cpina the commit you linked to is exactly what I was thinking.

**Comment by cpina** [comment]
> @cpina the commit you linked to is exactly what I was thinking.

`field.html` I guess that can be re-done a bit in Bootstrapv5. This is with a bit of conservative refactoring: https://github.com/cpina/django-crispy-forms/commit/e6a68a0d8bf8e3d501d6cd43d1d5c55c81060eab to avoid the duplicate. Unit test pass, it fixes this issue as well... I don't know if it's better or not than the simpler earlier fix.

Thanks for linking the related issue where I had explained the problem in December: I didn't actually remember looking into this with this detail!

(either way it needs a unit test and I'll do it)

**Comment by cpina** [comment]
Without a bigger refactoring in `field.html` I'm not happy with any solution to avoid code duplication in general speaking.

**Simple solution**
Go with the first solution: https://github.com/cpina/django-crispy-forms/commit/6b2df6c1e1b295d8bccb5213e895de30f1c48c1b with the unit test: https://github.com/cpina/django-crispy-forms/commit/57b9e1bc10b3fb3774dd0269b24c3c01c8af2b70 . I can open a PR with this.

**Intermediate solution**
We could move the 4 duplicated lines (in `field.html` and `prepended_appended_text.html`):
```python3

                {% if field|is_select and use_custom_control %}
                    {% crispy_field field 'class' 'custom-select' %}
                {% else %}
                    {% crispy_field field %}
                {% endif %}
```
Into a template. I can't think of a good name for this template.

**Bigger refactoring**
Refactor `field.html` and `prepended_appended_text.html` - but I'm thinking that this should be done for the Bootstrapv5 changes? I've been trying a few approaches without re-writing it but I wasn't convinced by any of them.

**Comment by cpina** [comment]
@smithdc1 : I haven't tried to improve the code since last weekend (I might have better ideas this weekend?), what do you think on going with the simple fix? https://github.com/django-crispy-forms/django-crispy-forms/compare/master...cpina:1111-custom-select_missing_prepended_appended_text_small_change?expand=1 and address that in a better way in BS5? I feel that this is a regression in the last release and I'm a bit more time-limited the next weeks for a bigger refactoring (compared with Dec-Jan)

**Comment by smithdc1** [comment]
"simple is better than complex"? 

I think let's get it in. It's a nice improvement and we can think about a better design later. It's not a massive issue either way in my view, but suspect I'm not experienced enough to judge!

Thanks again for your efforts here!

**Comment by smithdc1** [comment]
Hi @iGoosiatin -- Thanks again for the report. I've released version 1.11.1 which contains a fix for this. Do let us know if this doesn't work for you!

**Comment by iGoosiatin** [comment]
> Hi @iGoosiatin -- Thanks again for the report. I've released version 1.11.1 which contains a fix for this. Do let us know if this doesn't work for you!

Hi David,

Issue is fixed. Thanks you guys for suck a quick resolution!
